src/nspawn/nspawn.c

```diff
@@ -1029,20 +1029,35 @@ static int parse_argv(int argc, char *argv[]) {
                         assert_not_reached("Unhandled option");
                 }
 
+        if (getenv_bool("SYSTEMD_NSPAWN_SHARE_NS_IPC") > 0)
+                arg_share_ns &= ~CLONE_NEWIPC;
+        else
+                arg_share_ns |= CLONE_NEWIPC;
+
+        if (getenv_bool("SYSTEMD_NSPAWN_SHARE_NS_PID") > 0)
+                arg_share_ns &= ~CLONE_NEWPID;
+        else
+                arg_share_ns |= CLONE_NEWPID;
+
+        if (getenv_bool("SYSTEMD_NSPAWN_SHARE_NS_UTS") > 0)
+                arg_share_ns &= ~CLONE_NEWUTS;
+        else
+                arg_share_ns |= CLONE_NEWUTS;
```

Comment:
Given how complex this has become, I think it would be nice to add a function that that does this:

```c
static void parse_share_ns_env(const char *name, unsigned long mask) {
    int r;

        r = getenv_bool(name);
        if (r < 0)
                log_warning(....);
        else
                arg_share_ns = (args_share_ns & ~mask) | (r ? mask : 0);
}
```

or something like that. And then we can simply call this three times.
}